src/Chapter1.hs

```diff
@@ -541,8 +541,21 @@ value after "=" where the condition is true.
 
 Casual reminder about adding top-level type signatures for all functions :)
 -}
-
-mid x y z = error "mid: not implemented!"
+mid :: Int -> Int -> Int -> Int
+mid x y z
+    | (y < x) && (x < z) = x
+    | (x < y) && (y < z) = y
+    | (x < z) && (z < y) = z
+    | (z < x) && (x < y) = x
+    | (z < y) && (y < x) = y
+    | (y < z) && (z < x) = z
+    | (x == y) && (x < z) = x
+    | (x == y) && (z < x) = z
+    | (x == z) && (x < y) = x
+    | (x == z) && (y < z) = y
+    | (y == z) && (y < x) = y
+    | (y == z) && (x < y) = x
+    | otherwise = x
```

Comment:
Great solution! 👍🏼 

You can see, that some of the cases can be combined into one. For example,

```haskell
    | (y < x) && (x < z) = x
   | (x == y) && (x < z) = x
```
These two looks almost the same. You can write it this way:

```haskell
    | (y <= x) && (x < z) = x
```

And they will mean the same 🙂

Reply:
Thank you for the great code review 😄 

I've done the suggested changes so I can merge the PR with a better code.

A side note about the parenthesis, I thougth that the abs was only working because of it. Good to know that it's not necessary, thanks 🙂